### PR TITLE
Throw a more descriptive error about malformed XML.

### PR DESCRIPTION
**Pull request by jordandukart**

### CODE REVIEW

File: FoxmlDocument.php [hunk 1]
```diff
@@ -163,7 +163,9 @@ private function createInlineDocumentDatastream($ds) {
     $version = $this->createDatastreamVersionElement("{$ds->id}.0", $ds->label, $ds->mimetype, $ds->format);
     $content = $this->createDatastreamContentElement();
     $xml_dom = new DOMDocument();
-    $xml_dom->loadXML($ds->content);
+    if (!$xml_dom->loadXML($ds->content)) {
+      throw new RepositoryXmlError('Invalid XML datastream');
```
Comment (adam-vessey) [inline FoxmlDocument.php]: Would having the datastream ID as part of the message be a good idea?... I would like it, anyway. :P

Comment (jonathangreen) [inline FoxmlDocument.php]: Yah. PID and DSID would be nice. 

Also do we need to change the Exception class to allow the other arrangements to be NULL?

Comment (jonathangreen) [inline FoxmlDocument.php]: I also wonder if we could add a quick test to FoxmlDocument to test this out? Pass a bad datastream and see if the exception is thrown.

Comment (jordandukart) [inline FoxmlDocument.php]: Something along the lines of "ResponseText: RepositoryXmlError: DS-COMPOSITE-MODEL on usc:vroCModel contains invalid XML in FoxmlDocument->createInlineDocumentDatastream()" work for you folks?

Comment (jonathangreen) [inline FoxmlDocument.php]: Yah. That looks good to me!

**Pull request merged**